services/meta/data.go

```diff
@@ -369,16 +369,48 @@ func (data *Data) CreateShardGroup(database, policy string, timestamp time.Time)
 		return nil
 	}
 
+	startTime := timestamp.Truncate(rpi.ShardGroupDuration).UTC()
+	endTime := startTime.Add(rpi.ShardGroupDuration).UTC()
+	if endTime.After(time.Unix(0, models.MaxNanoTime)) {
+		// Shard group range is [start, end) so add one to the max time.
+		endTime = time.Unix(0, models.MaxNanoTime+1)
+	}
+
+	for i := range rpi.ShardGroups {
+		if rpi.ShardGroups[i].Deleted() {
+			continue
+		}
+		startI := rpi.ShardGroups[i].StartTime
+		endI := rpi.ShardGroups[i].EndTime
+		if rpi.ShardGroups[i].Truncated() {
+			endI = rpi.ShardGroups[i].TruncatedAt
+		}
```

Comment:
This basic code is copied and pasted a few times in the code. Would it make sense to add an `ShardGroupInfo.ActualEndTime()` method? I have a fear we'll find a bug one day due to one section of code not adjusting for `.TruncatedAt`